src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/CodeBackflower.cs

```diff
@@ -0,0 +1,244 @@
+        CancellationToken cancellationToken);
+}
+
+public enum BackflowAction
+{
+    CreatePatches,
+    ApplyPatches,
+}
+
+/// <summary>
+/// This class is responsible for taking changes done to a repo in the VMR and backflowing them into the repo.
+/// It only makes patches/changes locally, no other effects are done.
+/// </summary>
+public class CodeBackflower : IVmrBackflower
+{
+    private readonly IVmrInfo _vmrInfo;
+    private readonly ISourceManifest _sourceManifest;
+    private readonly ILocalGitRepo _localGitClient;
+    private readonly IVmrPatchHandler _vmrPatchHandler;
+    private readonly IWorkBranchFactory _workBranchFactory;
+    private readonly IFileSystem _fileSystem;
+    private readonly ILogger<CodeBackflower> _logger;
+
+    public CodeBackflower(
+        IVmrInfo vmrInfo,
+        ISourceManifest sourceManifest,
+        ILocalGitRepo localGitClient,
+        IVmrPatchHandler vmrPatchHandler,
+        IWorkBranchFactory workBranchFactory,
+        IFileSystem fileSystem,
+        ILogger<CodeBackflower> logger)
+    {
+        _vmrInfo = vmrInfo;
+        _sourceManifest = sourceManifest;
+        _localGitClient = localGitClient;
+        _vmrPatchHandler = vmrPatchHandler;
+        _workBranchFactory = workBranchFactory;
+        _fileSystem = fileSystem;
+        _logger = logger;
+    }
+
+    public async Task<List<VmrIngestionPatch>> CreateBackflowPatchesAsync(string mappingName, CancellationToken cancellationToken)
+    {
+        (string vmrSourceSha, string vmrTargetSha, IVersionedSourceComponent repo) = await GetMappingInformation(mappingName);
+        return await CreateBackflowPatchesAsync(repo, vmrSourceSha, vmrTargetSha, cancellationToken);
+    }
+
+    public async Task BackflowAsync(
+        BackflowAction action,
+        string mappingName,
+        NativePath repoDirectory,
+        IReadOnlyCollection<AdditionalRemote> additionalRemotes,
+        CancellationToken cancellationToken)
+    {
+        (string vmrSourceSha, string vmrTargetSha, IVersionedSourceComponent repo) = await GetMappingInformation(mappingName);
+
+        var patches = await CreateBackflowPatchesAsync(repo, vmrSourceSha, vmrTargetSha, cancellationToken);
+        if (patches.Count == 0)
+        {
+            return;
+        }
+
+        if (action == BackflowAction.CreatePatches)
+        {
+            var message = new StringBuilder();
+            message.AppendLine($"{patches.Count} patch{(patches.Count == 1 ? null : "es")} were created:");
+
+            var longestPath = patches.Max(p => p.Path.Length) + 4;
+
+            foreach (var patch in patches)
+            {
+                message.Append(patch.Path.PadRight(longestPath));
+                message.AppendLine(patch.ApplicationPath ?? string.Empty);
+            }
+
+            Console.WriteLine(message);
+            return;
+        }
+
+        _logger.LogInformation("Synchronizing {repo} from {repoSourceSha}", mappingName, repo.CommitSha);
```

Comment:
Can we put all this in a method and call it `ApplyPatches`? Or at least add a comment saying we're adding the patches here